frontend/src/Editor/Components/RadioButtonV2/RadioButtonV2.jsx

```diff
@@ -0,0 +1,287 @@
+import React, { useEffect, useMemo, useState, useRef } from 'react';
+import Label from '@/_ui/Label';
+import cx from 'classnames';
+import './radioButtonV2.scss';
+import Loader from '@/ToolJetUI/Loader/Loader';
+import { has, isObject } from 'lodash';
+
+export const RadioButtonV2 = ({
+  properties,
+  styles,
+  fireEvent,
+  setExposedVariable,
+  setExposedVariables,
+  darkMode,
+  componentName,
+  validate,
+  validation,
+}) => {
+  const { label, value, options, disabledState, advanced, schema, optionsLoadingState, loadingState } = properties;
+
+  const {
+    activeColor,
+    direction,
+    auto: labelAutoWidth,
+    labelWidth,
+    optionsTextColor,
+    borderColor,
+    switchOffBackgroundColor,
+    handleColor,
+    switchOnBackgroundColor,
+    labelColor,
+    alignment,
+  } = styles;
+
+  const isInitialRender = useRef(true);
+
+  const [checkedValue, setCheckedValue] = useState(advanced ? findDefaultItem(schema) : value);
+  const [visibility, setVisibility] = useState(properties.visibility);
+  const [isLoading, setIsLoading] = useState(loadingState);
+  const [isDisabled, setIsDisabled] = useState(disabledState);
+
+  const isMandatory = validation?.mandatory ?? false;
+  const validationData = validate(checkedValue);
+  const { isValid, validationError } = validationData;
+
+  const labelRef = useRef();
+  const radioBtnRef = useRef();
+
+  const selectOptions = useMemo(() => {
+    let _options = advanced ? schema : options;
+    if (Array.isArray(_options)) {
+      let _selectOptions = _options
+        .filter((data) => data?.visible ?? true)
+        .map((data) => ({
+          ...data,
+          label: data?.label,
+          value: data?.value,
+          isDisabled: data?.disable ?? false,
+        }));
+      return _selectOptions;
+    } else {
+      return [];
+    }
+  }, [advanced, schema, options]);
+
+  function findDefaultItem(optionSchema) {
+    if (!Array.isArray(optionSchema)) {
+      return undefined;
+    }
+    const foundItem = optionSchema?.find((item) => item?.default === true && item?.visible === true);
+    return foundItem?.value;
+  }
+
+  function onSelect(value) {
+    const _value = isObject(value) && has(value, 'value') ? value?.value : value;
+    setCheckedValue(_value);
```

Comment:
Validation will be inconsistent when onSelectionChange is fired as we are making changes to isValid in the next render.